util/docker/docker_cleanup.go

```diff
@@ -0,0 +1,76 @@
+package docker
+
+import (
+	"context"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+// Cleanup removes all containers, images, and volumes from a Docker instance
+// if a Docker daemon is listening on the default socket
+func Cleanup(ctx context.Context, logger grip.Journaler) error {
+	dockerClient, err := client.NewClientWithOpts()
+	if err != nil {
+		return errors.Wrap(err, "can't get Docker client")
+	}
+
+	info, err := dockerClient.Info(ctx)
+	if err != nil {
+		if client.IsErrConnectionFailed(err) {
+			// the daemon isn't running so there's nothing to clean up
+			logger.Info("Docker daemon isn't running")
+			return nil
+		}
+		return errors.Wrap(err, "can't get Docker info")
+	}
+
+	logger.Infof("Removing: %d containers (%d running, %d paused, %d stopped), %d images",
+		info.Containers,
+		info.ContainersRunning,
+		info.ContainersPaused,
+		info.ContainersStopped,
+		info.Images,
+	)
```

Comment:
I'd tend toward using message.Fields{} which might be more clear for this kind of structured data.

Reply:
done.